docs/v3/documentation/smart-contracts/transaction-fees/fees-low-level.md

```diff
@@ -233,18 +213,51 @@ will be translated into a few instructions which changes the order of elements o
 
 When the number of stack entries is substantial (10+), and they are actively used in different orders, stack operations fees may become non-negligible.
 
+
+## Forward fees
+
+Internal messages define an `ihr_fee` in Toncoins, which is subtracted from the value attached to the message and awarded to the validators of the destination shardchain if they include the message through the IHR mechanism. The `fwd_fee` is the original total forwarding fee paid for using the HR mechanism; it is automatically computed from the [24 and 25 configuration parameters](/v3/documentation/network/configs/blockchain-configs#param-24-and-25) and the size of the message at the time the message is generated. Note that the total value carried by a newly created internal outbound message equals the sum of the value, `ihr_fee`, and `fwd_fee`. This sum is deducted from the balance of the source account. Of these components, only the value is always credited to the destination account upon message delivery. The `fwd_fee` is collected by the validators on the HR path from the source to the destination, and the `ihr_fee` is either collected by the validators of the destination shardchain (if the message is delivered via IHR) or credited to the destination account.
+
+```cpp
+msg_fwd_fees = (lump_price
+             + ceil(
+                (bit_price * msg.bits + cell_price * msg.cells) / 2^16)
+             );
+
+ihr_fwd_fees = ceil((msg_fwd_fees * ihr_price_factor) / 2^16);
+
+total_fwd_fees = msg_fwd_fees + ihr_fwd_fees; // ihr_fwd_fees - is 0 for external messages
+```
+
 ## Action fee
 
-Action fee is deducted from the balance of the source account during processing action list which is perfomed after Computing phase.
-These are the actions that lead to pay fees:
+The action fee is deducted from the balance of the source account during the processing of the action list, which occurs after the Computing phase. Practically, the only action for which you pay an action fee is `SENDRAWMSG`. Other actions, such as `RAWRESERVE` or `SETCODE`, do not incur any fee during the action phase.
```

Comment:
Формально за SENDMSG тоже нужно платить action_fee, но я думаю это не столь обязательно уточнять

Reply:
What is `SENDMSG` action? I see only `SENDRAWMSG` both in TVM whitepapers - https://docs.ton.org/tvm.pdf,
and in official implementation transaction.cpp - https://github.com/ton-blockchain/ton/blob/master/crypto/block/transaction.cpp

Reply:
guess the misconception here is the existence of two `TVM instructions` - `SENDMSG` and `SENDRAWMSG`, that create output actions. Nevertheless, in relation to the c5 register and action phase, we are talking about `actions` 
themselves, one of which is the namesake `SENDRAWMSG`. Those are different entities. But if you think this will be a widespread misconception, we can add a WARN section describing that.

Reply:
Yes, my bad, indeed, I mixed up action and tvm instruction. I think there is no need to add anything here